File: 1053/1053.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var previousPermutationWithOneSwap = function(root) {
    if (!root) return 0;
    return 1 + Math.max(previousPermutationWithOneSwap(root.left), previousPermutationWithOneSwap(root.right));
};

File: 1053/1053.go
func previousPermutationWithOneSwap(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := previousPermutationWithOneSwap(root.Left)
	r := previousPermutationWithOneSwap(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}